index.js

```diff
@@ -1,30 +1,33 @@
 'use strict'
 
+const { readFile } = require('node:fs/promises')
 const { PassThrough } = require('node:stream')
 const path = require('node:path')
 const { fileURLToPath } = require('node:url')
 const { statSync } = require('node:fs')
-const { glob } = require('glob')
+const { Glob } = require('glob')
 const fp = require('fastify-plugin')
 const send = require('@fastify/send')
 const encodingNegotiator = require('@fastify/accept-negotiator')
 const contentDisposition = require('content-disposition')
 
 const dirList = require('./lib/dirList')
+const { generateHashes } = require('./lib/hash')
 
-const endForwardSlashRegex = /\/$/u
-const doubleForwardSlashRegex = /\/\//gu
 const asteriskRegex = /\*/gu
+const endForwardSlashRegex = /\/$/u
+
+const kFileHashes = Symbol('fileHashes')
 
+const defaultHashPath = path.join('.tmp', 'hashes.json')
 const supportedEncodings = ['br', 'gzip', 'deflate']
 send.mime.default_type = 'application/octet-stream'
 
 async function fastifyStatic (fastify, opts) {
   opts.root = normalizeRoot(opts.root)
   checkRootPathForErrors(fastify, opts.root)
 
-  const setHeaders = opts.setHeaders
-  if (setHeaders !== undefined && typeof setHeaders !== 'function') {
+  if (opts.setHeaders !== undefined && typeof opts.setHeaders !== 'function') {
```

Comment:
nit: unnecessary change

Reply:
Why was that const created? That was what I was trying to understand 🤣

But you're right it's not this PRs job to do that